[Navigation API] Not initialized for documents with opaque origins.

basuke opened the pull request:
#### 9b4d7bbfa91643a6a20ef7a3d00abc4d5532683f
<pre>
[Navigation API] Not initialized for documents with opaque origins.
<a href="https://bugs.webkit.org/show_bug.cgi?id=297371">https://bugs.webkit.org/show_bug.cgi?id=297371</a>
<a href="https://rdar.apple.com/158265130">rdar://158265130</a>

Reviewed by Tim Nguyen.

Navigation API was incorrectly skipping initialization for documents with opaque origins
(e.g., sandboxed iframes), causing tests to fail. The spec requires Navigation API to be
initialized for all documents, but return appropriate disabled state for opaque origins.

Fixed by:
- Removing opaque origin check from FrameLoader initialization condition.
- Navigation API methods already correctly return disabled state via
  hasEntriesAndEventsDisabled() for opaque origins per spec.
- Copied failing test to http/wpt/navigation-api/ with inlined dependencies
  to work around WebKit test server CORS restrictions for null origins
- Extracted complex traversal dispatch conditions into reusable helper methods

* LayoutTests/TestExpectations:
* LayoutTests/http/wpt/navigation-api/navigation-history-entry/opaque-origin-expected.txt: Added.
* LayoutTests/http/wpt/navigation-api/navigation-history-entry/opaque-origin.html: Added.
* LayoutTests/http/wpt/navigation-api/navigation-history-entry/resources/opaque-origin-page.html: Added.
* LayoutTests/http/wpt/navigation-api/navigation-methods/return-value/back-forward-opaque-origin-expected.txt: Added.
* LayoutTests/http/wpt/navigation-api/navigation-methods/return-value/back-forward-opaque-origin.html: Added.
* LayoutTests/http/wpt/navigation-api/navigation-methods/return-value/resources/back-forward-opaque-origin-page.html: Added.
* Source/WebCore/loader/FrameLoader.cpp:
(WebCore::FrameLoader::shouldDispatchNavigateEventForHistoryTraversal):
(WebCore::FrameLoader::loadItem):
* Source/WebCore/loader/FrameLoader.h:

Canonical link: <a href="https://commits.webkit.org/298688@main">https://commits.webkit.org/298688@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/152678c73f1a7226747b005094dc3f19252412cb

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/116364 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/36025 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/26575 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/122421 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/66925 "Built successfully") | ⏳ 🛠 ios-apple 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/118253 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/36719 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/44613 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/88387 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/42862 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | ⏳ 🛠 mac-apple 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/119313 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/29283 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/104397 "Passed tests") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/68821 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/28365 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/22500 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/66102 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/98665 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/22661 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/125570 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/43258 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/32487 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/97093 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 vision-sim~~](https://ews-build.webkit.org/#/builders/121/builds/43623 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/100603 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/96888 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/24649 "Built successfully and passed tests") | [⏳ 🧪 vision-wk2 ](https://ews-build.webkit.org/#/builders/visionOS-2-Simulator-WK2-Tests-EWS "Waiting in queue, processing has not started yet") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/20072 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/39212 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/43146 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/48737 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/42612 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/45952 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/44317 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/855bf30a75fdcf036793eed9efd4af9627f643a8. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/49367

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/152678c73f1a7226747b005094dc3f19252412cb. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/49367

basuke committed:
[Navigation API] Not initialized for documents with opaque origins.
https://bugs.webkit.org/show_bug.cgi?id=297371
rdar://158265130

Reviewed by Tim Nguyen.

Navigation API was incorrectly skipping initialization for documents with opaque origins
(e.g., sandboxed iframes), causing tests to fail. The spec requires Navigation API to be
initialized for all documents, but return appropriate disabled state for opaque origins.

Fixed by:
- Removing opaque origin check from FrameLoader initialization condition.
- Navigation API methods already correctly return disabled state via
  hasEntriesAndEventsDisabled() for opaque origins per spec.
- Copied failing test to http/wpt/navigation-api/ with inlined dependencies
  to work around WebKit test server CORS restrictions for null origins
- Extracted complex traversal dispatch conditions into reusable helper methods

* LayoutTests/TestExpectations:
* LayoutTests/http/wpt/navigation-api/navigation-history-entry/opaque-origin-expected.txt: Added.
* LayoutTests/http/wpt/navigation-api/navigation-history-entry/opaque-origin.html: Added.
* LayoutTests/http/wpt/navigation-api/navigation-history-entry/resources/opaque-origin-page.html: Added.
* LayoutTests/http/wpt/navigation-api/navigation-methods/return-value/back-forward-opaque-origin-expected.txt: Added.
* LayoutTests/http/wpt/navigation-api/navigation-methods/return-value/back-forward-opaque-origin.html: Added.
* LayoutTests/http/wpt/navigation-api/navigation-methods/return-value/resources/back-forward-opaque-origin-page.html: Added.
* Source/WebCore/loader/FrameLoader.cpp:
(WebCore::FrameLoader::shouldDispatchNavigateEventForHistoryTraversal):
(WebCore::FrameLoader::loadItem):
* Source/WebCore/loader/FrameLoader.h:

Canonical link: https://commits.webkit.org/298688@main
